### PR TITLE
Revise Link Check Action

### PR DESCRIPTION
**Pull request by kingthorin**
- package.json > Remove 'link-check' script.
- md-link-check.yml > Only run when *.md files are included. Add 'Changed Files Exporter' step. Change 'Run link check' step to run CLI actions in-line.

Relevant test usage can be seen here in commits from "Test 37" onward: https://github.com/kingthorin/testwstg/pull/1
(Note the test repo will be deleted after this PR is approved/merged.)

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,10 +7,6 @@
   "devDependencies": {
     "markdownlint-cli": "^0.19.0"
   },
-  "scripts": {
-    "test": "markdownlint .  --ignore node_modules",
```
Comment (rejahrehim) [inline package.json]: Removing `test` will break lint checker.

File: .github/workflows/md-link-check.yml [hunk 2]
```diff
@@ -20,8 +22,20 @@ jobs:
         node-version: 12.x
     - name: Install dependencies
       run: npm install -g markdown-link-check
+    - name: Changed Files Exporter
+      id: files
+      uses: futuratrepadeira/changed-files@v3.0.0
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
     - name: Run link check
```
Comment (rejahrehim) [inline .github/workflows/md-link-check.yml]: Sorry I missed this before.

Can we make this step conditional to make sure to trigger only a change detects in markdown files? As of now it runs and fails.  

https://github.com/OWASP/wstg/runs/459370407?check_suite_focus=true

Comment (kingthorin) [reply .github/workflows/md-link-check.yml]: That action was against cf96722, which was before the path restriction `**.md` was added.

Comment (rejahrehim) [reply .github/workflows/md-link-check.yml]: Ohh.. Okay.. Then all fine..

Comment (kingthorin) [reply .github/workflows/md-link-check.yml]: Actually I wonder if we should add that path restriction to the lint workflow as well? (As you can see below linting did run on this PR (on commit ccda9d8 ), but that's actually kind of pointless.)

Comment (rejahrehim) [reply .github/workflows/md-link-check.yml]: Sure, That I will do.  :)

**Pull request merged**